src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -277,23 +295,23 @@ public final Builder name(String name) {
 		 * Set the options to use for configuring {@link ConnectionProvider} acquire timeout.
 		 * Default to DEFAULT_POOL_ACQUIRE_TIMEOUT.
 		 *
-		 * @param acquireTimeout value must be a positive.
+		 * @param acquireTimeout the maximum time after which a pending acquire
+		 * must complete or the {@link TimeoutException} will be thrown (resolution: ms)
 		 * @return {@literal this}
-		 * @throws IllegalArgumentException if acquireTimeout is negative
+		 * @throws NullPointerException if acquireTimeout is null
 		 */
-		public final Builder acquireTimeout(long acquireTimeout) {
-			if (acquireTimeout < 0) {
-				throw new IllegalArgumentException("Acquire Timeout value must be positive");
-			}
-			this.acquireTimeout = acquireTimeout;
+		public final Builder acquireTimeout(Duration acquireTimeout) {
+			this.acquireTimeout = Objects.requireNonNull(acquireTimeout, "acquireTimeout");
 			return this;
 		}
 
 		/**
 		 * Set the options to use for configuring {@link ConnectionProvider} maximum connections.
 		 * Default to DEFAULT_POOL_MAX_CONNECTIONS.
+		 * When invoked with MAX_CONNECTIONS_ELASTIC an elastic ConnectionProvider will be created
+		 * and acquireTimeout will be set automatically to ACQUIRE_TIMEOUT_NEVER_WAIT.
```

Comment:
These could  `{@link ...}` to the constants.

Reply:
done